blog/2023-06-20-Deployment-vs-Release.md

```diff
@@ -0,0 +1,56 @@
+
+**Canary Deployment**: With canary a new version is initially released to a small subset of servers or users. If everything works well, the new version is progressively rolled out to the rest of the servers or users.
+
+**A/B Testing**: While similar to a Canary deployment, in A/B testing the performance of both versions is compared in order to decide whether the new version is beneficial or not.
+
+**Feature Flags (or Feature Toggles)**: Feature flags allow developers to control the visibility and functionality of a specific feature in a live environment. This means they can turn features on or off, enable them for specific user groups, or roll them out progressively. This deployment strategy enables teams to test new features in the production environment without affecting the entire user base. If an issue arises with the new feature, it can be toggled off immediately, reducing the impact on users. Feature flags provide a safety net and offer flexibility for both deployments and releases.
+
+Deployment and release practices often hit plenty common challenges that can have an impact on their success, especially those practices that have grown organically over time rather than through explicit direction and planning. Recognising these obstacles is the first step towards mitigating them and, towards enhancing the efficiency of these processes. 
+For example, deploying a new version could be a rigid and laborious process. Changes, often require a significant amount of effort to implement, especially when dependencies or configurations are separated from the artefact, managed by different teams or third parties. 
+Then we have the human error factor, brought on by manual deployment processes. The lack of automation can lead to mistakes in configuration, incorrect file transfers, or forgotten/overlooked steps, resulting in failed deployments and service disruptions. This underscores the mantra in modern development: automate everything!
+In deployment models where changes accumulate behind infrequent release events, coordinating all necessary stakeholders from developers and testers, operations ans support, to marketing and leadership - it becomes an overly complex, challenging and unrepeatable process. This usually creates a lack of trust across the teams involved, but an issue that can be alleviated through good practices and open communication.
+Finally, when users discover issues or bugs that slipped past testing, experience unexpected downtime during deployments, or suffer from reduced performance. All these factors can lead to user dissatisfaction.This can create churn in your product and cause loss of trust within your customer.
```

Comment:
```suggestion
Finally, when users discover issues or bugs that slipped past testing, experience unexpected downtime during deployments, or suffer from reduced performance. All these factors can lead to user dissatisfaction. This can create churn in your product and cause a loss of trust within your customer.
```